frontend/src/components/Comments/Comments.jsx

```diff
@@ -0,0 +1,97 @@
+import "./Comments.css";
+import { useState, useEffect, useContext } from "react";
+import axios from "axios";
+import { UserContext } from "../App/App";
+import { formatDateTime, capitalize } from "../../utils";
+
+export default function Comments({ ticker }) {
+    const { setError } = useContext(UserContext);
+    const [comments, setComments] = useState([]);
+    const [commentInput, setCommentInput] = useState("");
+
+    const fetchComments = async () => {
+        try {
+            const response = await axios.get(
+                `http://localhost:3000/comments/${ticker}`,
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
+                setComments(response.data.comments);
+            }
+
+            if (response.status === 404) {
+                setError(response.data.error);
+            }
+
+            if (response.status === 500) {
+                setError(`${response.statusText}: Please try again later.`);
+            }
+        } catch (error) {
+            setError(`${error.message}: Please try again later.`);
+        }
+    };
+
+    useEffect(() => {
+        fetchComments();
+    }, []);
+
+    const handleSubmit = async (event) => {
+        event.preventDefault();
+
+        try {
+            const commentData = {
+                ticker,
+                content: commentInput
+            };
+
+            const response = await axios.post(
+                "http://localhost:3000/comments",
+                commentData,
+                { withCredentials: true, validateStatus: () => true }
+            );
+
+            if (response.status === 200) {
+                fetchComments();
```

Comment:
This additional network call can be avoided with an "optimistic update". We have a successful response, so we assume the comment went in to the DB, thus, we can add it to the current state.